sqldelight-compiler/src/test/kotlin/com/squareup/sqldelight/core/QueriesTypeTest.kt

```diff
@@ -132,4 +134,126 @@ class QueriesTypeTest {
+      |);
+      |
+      |insertData:
+      |INSERT INTO data
+      |VALUES (?, ?);
+      |
+      |selectForId:
+      |SELECT *
+      |FROM data
+      |WHERE id = ?;
+    """.trimMargin(), temporaryFolder, fileName = "Data.sq")
+
+        val select = result.compiledFile.namedQueries.first()
+        val insert = result.compiledFile.namedMutators.first()
+        assertThat(result.errors).isEmpty()
+
+        val dataQueries = File(result.outputDirectory, "com/example/testmodule/TestDatabaseImpl.kt")
+        assertThat(result.compilerOutput).containsKey(dataQueries)
+        assertThat(result.compilerOutput[dataQueries].toString()).isEqualTo("""
+      |package com.example.testmodule
+      |
+      |import com.example.Data
+      |import com.example.DataQueries
+      |import com.example.TestDatabase
+      |import com.squareup.sqldelight.Query
+      |import com.squareup.sqldelight.TransacterImpl
+      |import com.squareup.sqldelight.db.SqlCursor
+      |import com.squareup.sqldelight.db.SqlDriver
+      |import com.squareup.sqldelight.internal.copyOnWriteList
+      |import kotlin.Any
+      |import kotlin.Int
+      |import kotlin.Long
+      |import kotlin.String
+      |import kotlin.collections.List
+      |import kotlin.collections.MutableList
+      |import kotlin.jvm.JvmField
+      |import kotlin.reflect.KClass
+      |
+      |internal val KClass<TestDatabase>.schema: SqlDriver.Schema
+      |  get() = TestDatabaseImpl.Schema
+      |
+      |internal fun KClass<TestDatabase>.newInstance(driver: SqlDriver): TestDatabase =
+      |    TestDatabaseImpl(driver)
+      |
+      |private class TestDatabaseImpl(
+      |  driver: SqlDriver
+      |) : TransacterImpl(driver), TestDatabase {
+      |  override val dataQueries: DataQueriesImpl = DataQueriesImpl(this, driver)
+      |
+      |  object Schema : SqlDriver.Schema {
+      |    override val version: Int
+      |      get() = 1
+      |
+      |    override fun create(driver: SqlDriver) {
+      |      driver.execute(null, ""${'"'}
+      |          |CREATE VIRTUAL TABLE data USING fts5(
+      |          |  id,
+      |          |  value
+      |          |)
+      |          ""${'"'}.trimMargin(), 0)
+      |    }
+      |
+      |    override fun migrate(
+      |      driver: SqlDriver,
+      |      oldVersion: Int,
+      |      newVersion: Int
+      |    ) {
+      |    }
+      |  }
+      |}
+      |
+      |private class DataQueriesImpl(
+      |  private val database: TestDatabaseImpl,
+      |  private val driver: SqlDriver
+      |) : TransacterImpl(driver), DataQueries {
+      |  internal val selectForId: MutableList<Query<*>> = copyOnWriteList()
+      |
+      |  override fun <T : Any> selectForId(id: Long, mapper: (id: Long, value: List?) -> T): Query<T> =
+      |      SelectForIdQuery(id) { cursor ->
+      |    mapper(
+      |      cursor.getLong(0)!!,
+      |      cursor.getString(1)?.let(database.dataAdapter.valueAdapter::decode)
```

Comment:
Which is accessed here (this part looks good)

I think somewhere there’s a method that checks to see if a table needs an adapter generated, that method probably needs a similar fix